fcrepo-kernel-modeshape/src/test/java/org/fcrepo/kernel/modeshape/UrlBinaryTest.java

```diff
@@ -0,0 +1,285 @@
+
+import java.io.InputStream;
+import java.net.URI;
+import javax.jcr.Binary;
+import javax.jcr.Node;
+import javax.jcr.Property;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import javax.jcr.Value;
+import javax.jcr.nodetype.NodeType;
+
+import org.fcrepo.kernel.api.exception.InvalidChecksumException;
+import org.fcrepo.kernel.api.models.FedoraBinary;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+
+/**
+ * @author bbpennel
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class UrlBinaryTest {
+
+    private static final String DS_ID = "testDs";
+
+    private static final String EXPECTED_CONTENT = "test content";
+
+    private FedoraBinary testObj;
+
+    private String mimeType;
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(options().dynamicPort());
+
+    private String fileUrl;
+
+    @Mock
+    private Session mockSession;
+
+    @Mock
+    private Property mimeTypeProperty;
+
+    @Mock
+    private Property proxyURLProperty;
+
+    @Mock
+    private Property redirectURLProperty;
+
+    @Mock
+    private Property mockProperty;
+
+    @Mock
+    private Value mockValue;
+
+    @Mock
+    private Value mockURIValue;
+
+    @Mock
+    private Node mockDescNode, mockContent, mockParentNode;
+
+    @Mock
+    private NodeType mockDescNodeType;
+
+    @Mock
+    private InputStream mockStream;
+
+    @Before
+    public void setUp() throws Exception {
+        final NodeType[] nodeTypes = new NodeType[] { mockDescNodeType };
+
+        stubFor(get(urlEqualTo("/file.txt"))
+                .willReturn(aResponse()
+                        .withHeader("Content-Type", "text/plain")
+                        .withBody(EXPECTED_CONTENT)));
+        fileUrl = "http://localhost:" + wireMockRule.port() + "/file.txt";
+
+        mimeType = "text/plain";
+
+        when(mockDescNodeType.getName()).thenReturn(FEDORA_NON_RDF_SOURCE_DESCRIPTION);
+        when(mockDescNode.getMixinNodeTypes()).thenReturn(nodeTypes);
+        when(mockDescNode.getParent()).thenReturn(mockContent);
+        when(mockContent.getSession()).thenReturn(mockSession);
+        when(mockContent.isNodeType(FEDORA_BINARY)).thenReturn(true);
+        when(mockContent.getParent()).thenReturn(mockParentNode);
+        when(mockContent.getNode(FEDORA_DESCRIPTION)).thenReturn(mockDescNode);
+        when(mockDescNode.setProperty(anyString(), any(Binary.class))).thenReturn(mockProperty);
+
+        final NodeType mockNodeType = mock(NodeType.class);
+        when(mockNodeType.getName()).thenReturn("nt:versionedFile");
+        when(mockContent.getPrimaryNodeType()).thenReturn(mockNodeType);
+
+        testObj = new UrlBinary(mockContent);
+    }
+
+    @Test
+    public void testSetContent() throws Exception {
```

Comment:
I don't think we need any setContent tests for URLBinary, other than maybe to verify that it throws an exception (see my comments in URLBinary)

Also, there aren't any tests for setExternalContent currently. I don't care too much if those get added or not since setExternalContent gets tested in ITs, so I'll leave if it up to you (or others) whether that's an issue.

Reply:
Hi Ben, I just added a few that explicitly call setExternalContent, since I realized that I hadn't added any specific tests for that (though the setContent was covered quite well).